src/resolve/resolved-dns-packet.c

```diff
@@ -47,13 +50,14 @@ int dns_packet_new(DnsPacket **ret, DnsProtocol protocol, size_t mtu) {
 
         assert(ret);
 
-        if (mtu <= UDP_PACKET_HEADER_SIZE)
+        /* When dns_packet_new() is called with mtu == 0, allocate more than the
+         * absolute minimum (which is the dns packet header size), to avoid
+         * resizing immediately again after appending the first data to the packet.
+         */
+        if (mtu < UDP_PACKET_HEADER_SIZE)
```

Comment:
@benjarobin the reason it's called MTU is that when reading a DNS UDP packet off the network, and know the MTU of the iface, then we know how large the packet can grow to at max, and then use that to size the allocation from the beginning, so that we don't have to reallocate later on.